src/grt/src/fastroute/src/utility.cpp

```diff
@@ -1882,9 +1882,39 @@ void FastRouteCore::copyRS(void)
 
 void FastRouteCore::copyBR(void)
 {
-  int i, j, edgeID, numEdges, numNodes, min_y, min_x;
+  int i, j, edgeID, numEdges, numNodes, min_y, min_x, edgeCost;
 
   if (!sttrees_bk_.empty()) {
+    // Reduce usage with last routes before update
+    for (const int& netID : net_ids_) {
+      numEdges = sttrees_[netID].num_edges();
+      edgeCost = nets_[netID]->getEdgeCost();
+
+      for (edgeID = 0; edgeID < numEdges; edgeID++) {
+        if (sttrees_[netID].edges[edgeID].len > 0) {
+          const std::vector<int16_t>& gridsX
+              = sttrees_[netID].edges[edgeID].route.gridsX;
+          const std::vector<int16_t>& gridsY
+              = sttrees_[netID].edges[edgeID].route.gridsY;
+          for (i = 0; i < sttrees_[netID].edges[edgeID].route.routelen; i++) {
```

Comment:
You can refactor `sttrees_[netID].edges[edgeID].route.routelen` to:

`TreeEdge& edge = sttrees_[netID].edges[edgeID];`

Then use the `edge` var inside the `for` clause:

`for (i = 0; i < edge.route.routelen; i++) {`